### PR TITLE
[WIP] ci: setup GitHub actions for windows

### PR DESCRIPTION
**Pull request by vigoux**
Use GitHub actions for windows CI, as I have encoutered way to many issues with AppVeyor.

For now, the build does not complete because of an error using `perl`.
This does not change the actual build scripts at all for now.

cc @mhinz

### CODE REVIEW

File: ci/build.ps1 [hunk 1]
```diff
@@ -36,20 +35,20 @@ $scoop = (New-Object System.Net.WebClient).DownloadString('https://get.scoop.sh'
   Invoke-Expression $scoop
 }
 
-scoop install perl
-perl --version
-cpanm.bat --version
+# scoop install perl@5.30.3.1
```
Comment (janlazo) [inline ci/build.ps1:38]: What's the error?

Comment (vigoux) [reply ci/build.ps1:38]: Something about overwriting `Config::STORE`.
But I'll try to reimolement the script as a GH workflow instead, to see how it goes.

**Comment by janlazo** [comment]
Any progess? It's fine if only a MSVC build passes. Appveyor hasn't updated all of its images to resolve mingw issues.

**Comment by vigoux** [comment]
Nope, no time to invest in this unfortunately... But feel free to try and take that as a starting point for something else.

I'll try to find some time later but can't guarantee anything...

**Comment by janlazo** [comment]
Obsoleted by https://github.com/neovim/neovim/pull/13540

**Pull request closed without merging**